repository-data/webfiles/src/main/resources/site/freemarker/common/api-catalogue.ftl

```diff
@@ -49,23 +50,38 @@
 
         <#if alphabetical_hash??>
             <div class="grid-row">
-                <div class="column column--one-third page-block page-block--sidebar">
+                <div class="column column--one-third page-block page-block--sidebar sticky sticky--top">
                     <@alphabeticalFilterNav alphabetical_hash></@alphabeticalFilterNav>
 
                     <#if filtersModel?? && !filtersModel.isEmpty()>
                         <div class="article-section-nav-wrapper">
-                            <div class="article-section-nav">
-                                <h2 class="article-section-nav__title">Filters</h2>
-                                <nav class="filters">
+                            <div class="article-section-nav filters">
+
+                                <h2 class="article-section-nav__title">
+                                    <span class="filter-head__title">Filters</span>
+                                    <@hst.renderURL var="resetUrl"/>
```

Comment:
could you explain what this macro does?

Reply:
It saves the value of the URL of the current page in a template variable whose name is given via `var`. This variable can then be used to render the URL, say, as the `<a href=` value (which is what we do in the next line with the `resetUrl` variable).

The URL contains the domain and path of the page but does not contain any query parameters. Given that we pass selected filters as query parameters, opening the page through a URL with no parameters has the effect of 'unsetting' any filters that may have been set - which is what this implementation of the _Reset_ button does.

It's part of the 'standard' tag library included by Bloomreach; they are actually JSP tags but can be used from Freemarker. For reference:

- [Bloomreach page that mentions the tags](https://documentation.bloomreach.com/14/library/concepts/web-application/hst2-tag-library.htmlframework).
- [Full collection of available tags](https://javadoc.onehippo.org/13.0/tlddocs/)
- [renderUrl tag](https://javadoc.onehippo.org/13.0/tlddocs/hst-core-tags/renderURL.html)